DataFormats/ForwardDetId/src/HGCSiliconDetId.cc

```diff
@@ -0,0 +1,62 @@
+#include "DataFormats/ForwardDetId/interface/HGCSiliconDetId.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include <ostream>
+#include <iostream>
+
+const HGCSiliconDetId HGCSiliconDetId::Undefined(HGCalEE,0,0,0,0,0,0,0);
+
+HGCSiliconDetId::HGCSiliconDetId() : DetId() {
+}
+
+HGCSiliconDetId::HGCSiliconDetId(uint32_t rawid) : DetId(rawid) {
+}
+
+HGCSiliconDetId::HGCSiliconDetId(DetId::Detector det, int zp, int type, 
+				 int layer, int waferU, int waferV, int cellU,
+				 int cellV) : DetId(det,ForwardEmpty) {  
+
+  int waferUabs(std::abs(waferU)), waferVabs(std::abs(waferV));
+  int waferUsign = (waferU >= 0) ? 0 : 1;
+  int waferVsign = (waferV >= 0) ? 0 : 1;
+  int zside      = (zp < 0) ? 1 : 0;
+  id_ |= ((cellU     & kHGCalCellUMask) << kHGCalCellUOffset);
+  id_ |= ((cellV     & kHGCalCellVMask) << kHGCalCellVOffset);
+  id_ |= ((waferUabs & kHGCalWaferUMask) << kHGCalWaferUOffset);
+  id_ |= ((waferUsign& kHGCalWaferUSignMask) << kHGCalWaferUSignOffset);
+  id_ |= ((waferVabs & kHGCalWaferVMask) << kHGCalWaferVOffset);
+  id_ |= ((waferVsign& kHGCalWaferVSignMask) << kHGCalWaferVSignOffset);
+  id_ |= ((layer     & kHGCalLayerMask) << kHGCalLayerOffset);
+  id_ |= ((zside     & kHGCalZsideMask) << kHGCalZsideOffset);
+  id_ |= ((type      & kHGCalTypeMask)  << kHGCalTypeOffset);
+}
+
+HGCSiliconDetId::HGCSiliconDetId(const DetId& gen) {
+  if (!gen.null()) {
+    if ((gen.det()!=HGCalEE) && (gen.det()!=HGCalHSi)) {
+      throw cms::Exception("Invalid DetId") << "Cannot initialize HGCSiliconDetId from " << std::hex << gen.rawId() << std::dec; 
```

Comment:
What happens if you try to construct one of these from an old `HGCEEDetId`?

Reply:
The entire design is totally different. When we move from rectangular to hexagon and then with a new way of defining geometry etc we do not need to convert one from other

Reply:
But what if it happens by accident...

Reply:
it will give error get, subset won't match

Reply:
Ah, I didn't look closely enough to realize that this was using the previously-proposed approach of keeping the subdet info in the det bits. I think that should be okay then.